### PR TITLE
Removes user executor

### PR DESCRIPTION
**Pull request by vbekiaris**
Removes user executor and associated config, as discussed [here](https://github.com/hazelcast/hazelcast/pull/16154#issuecomment-563109717)

### CODE REVIEW

**Comment by vbekiaris** [comment]
run-lab-run

**Comment by vbekiaris** [comment]
run-lab-run

File: hazelcast/src/main/java/com/hazelcast/client/impl/spi/ClientContext.java [hunk 1]
```diff
@@ -170,7 +171,7 @@ public ClientTransactionManagerService getTransactionManager() {
         return transactionManager;
     }
 
-    public ClientExecutionService getExecutionService() {
+    public TaskScheduler getExecutionService() {
```
Comment (sancar) [inline hazelcast/src/main/java/com/hazelcast/client/impl/spi/ClientContext.java]: I think we can remove `ClientExecutionService` interface completely and also change the method name to `getTaskScheduler` . There seems to be no use left for ClientExecutionService interface anymore.

File: hazelcast/src/test/java/com/hazelcast/client/test/TestClientRegistry.java [hunk 2]
```diff
@@ -313,7 +313,7 @@ public String toString() {
                 }
             }));
             executor.shutdownIncoming();
-
+            System.out.println(">>>> >>>>>");
```
Comment (kwart) [inline hazelcast/src/test/java/com/hazelcast/client/test/TestClientRegistry.java]: Leftover?

File: hazelcast/src/test/java/com/hazelcast/client/test/TestClientRegistry.java [hunk 3]
```diff
@@ -329,6 +329,7 @@ public void run() {
                 }
             });
             executor.shutdownOutgoing();
+            System.out.println("<<<< <<<<");
```
Comment (kwart) [inline hazelcast/src/test/java/com/hazelcast/client/test/TestClientRegistry.java]: Leftover?

**Comment by vbekiaris** [comment]
PR buidler failed with known issue #16099

**Pull request merged**